snippets/project-tree.ttl

```diff
@@ -0,0 +1,69 @@
+PREFIX st: <http://www.w3.org/ns/shapetrees#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX ex: <http://www.example.com/ns/ex#>
+
+<> st:describedBy ex:project-en, ex:project-es, ex:project-nl, ex:project-ko .
+
+<#ProjectTree>
+  a st:ShapeTree ;
+  st:expectsType st:Container ;
+  st:shape ex:ProjectShape ;
+  st:contains <#MilestoneTree> .
+
+<#MilestoneTree>
+  a st:ShapeTree ;
+  st:expectsType st:Container ;
+  st:shape ex:MilestoneShape ;
+  st:contains <#TaskTree>, <#IssueTree> .
+
+<#TaskTree>
+  a st:ShapeTree ;
+  st:expectsType st:Container ;
+  st:shape ex:TaskShape ;
+  st:contains st:NonRDFResourceTree .
```

Comment:
There are several refs to a ShapeTree called `st:NonRDFResourceTree`. I changed them to <#NonRDFResourceTree> and instantiated it in [shex-simple](http://shex.io/webapps/shex.js/doc/shex-simple.html?schema=BASE%20%3Chttp%3A%2F%2Fwww.w3.org%2Fns%2Fshapetrees%3E%0APREFIX%20st%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2Fns%2Fshapetrees%23%3E%0APREFIX%20rdfs%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0APREFIX%20xsd%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23%3E%0APREFIX%20skos%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2004%2F02%2Fskos%2Fcore%23%3E%0A%0A%3C%23ShapeTreeShape%3E%20%7B%0A%20%20a%20%5Bst%3AShapeTree%5D%20%3B%0A%20%20(%0A%20%20%20%20st%3AexpectsType%20%5Bst%3AContainer%5D%20%3B%0A%20%20%20%20st%3Acontains%20%40%3C%23ShapeTreeShape%3E%20*%20%3B%0A%20%20%20%20%7C%0A%20%20%20%20st%3AexpectsType%20%5Bst%3AResource%20st%3ANonRDFResource%5D%0A%20%20)%20%3B%0A%20%20rdfs%3Alabel%20xsd%3Astring%20%3F%20%3B%0A%20%20st%3Areferences%20%40%3C%23ReferenceShape%3E%20*%20%3B%0A%20%20st%3Ashape%20IRI%20%3F%0A%7D%0A%0A%3C%23ReferenceShape%3E%20%7B%0A%20%20st%3AreferencesShapeTree%20IRI%20%3B%0A%20%20(%20st%3AviaShapePath%20xsd%3Astring%0A%20%20%20%20%7C%0A%20%20%20%20st%3AviaPredicate%20IRI%20)%0A%7D%0A%0A%3C%23ManagerShape%3E%20%7B%0A%20%20a%20%5Bst%3AManager%5D%20%3B%0A%20%20st%3AhasAssignment%20%40%3C%23AssignmentShape%3E%20%2B%0A%7D%0A%0A%3C%23AssignmentShape%3E%20%7B%0A%20%20a%20%5Bst%3AAssignment%5D%20%3B%0A%20%20st%3Aassigns%20IRI%20%3B%0A%20%20st%3Amanages%20IRI%20%3B%0A%20%20st%3AhasRootAssignment%20IRI%20%3B%0A%20%20(%20st%3AfocusNode%20IRI%20%3B%0A%20%20%20%20st%3Ashape%20IRI%20)%20%3F%0A%7D%0A%0A%3C%23DescriptionSetShape%3E%20%7B%0A%20%20a%20%5Bst%3ADescriptionSet%5D%20%3B%0A%20%20st%3AusesLanguage%20xsd%3Alanguage%0A%7D%0A%0A%3C%23DescriptionShape%3E%20%7B%0A%20%20a%20%5Bst%3ADescription%5D%20%3B%0A%20%20st%3AinDescriptionSet%20%40%3C%23DescriptionSetShape%3E%20%3B%0A%20%20st%3Adescribes%20IRI%20%3B%0A%20%20skos%3AprefLabel%20xsd%3Astring%20%3B%0A%20%20skos%3Adefinition%20xsd%3Astring%20%3F%20%3B%0A%20%20skos%3Anarrower%20IRI%20%3F%20%3B%0A%20%20skos%3Abroader%20IRI%3F%20%3B%0A%7D&data=BASE%20%3Chttp%3A%2F%2Fa.example%2Fstz%3E%0APREFIX%20st%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2Fns%2Fshapetrees%23%3E%0APREFIX%20rdfs%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0APREFIX%20ex%3A%20%3Chttp%3A%2F%2Fwww.example.com%2Fns%2Fex%23%3E%0A%0A%3C%3E%20st%3AdescribedBy%20ex%3Aproject-en%2C%20ex%3Aproject-es%2C%20ex%3Aproject-nl%2C%20ex%3Aproject-ko%20.%0A%0A%3C%23ProjectTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3AProjectShape%20%3B%0A%20%20st%3Acontains%20%3C%23MilestoneTree%3E%20.%0A%0A%3C%23MilestoneTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3AMilestoneShape%20%3B%0A%20%20st%3Acontains%20%3C%23TaskTree%3E%2C%20%3C%23IssueTree%3E%20.%0A%0A%3C%23TaskTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3ATaskShape%20%3B%0A%20%20st%3Acontains%20%3C%23NonRDFResourceTree%3E%20.%0A%0A%3C%23IssueTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3AIssueShape%20%3B%0A%20%20st%3Acontains%20%3C%23NonRDFResourceTree%3E%20.%0A%0A%3C%23VirtualProjectTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AResource%20%3B%0A%20%20st%3Ashape%20ex%3AProjectShape%20%3B%0A%20%20st%3Areferences%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20%3C%23VirtualMilestoneTree%3E%20%3B%0A%20%20%20%20st%3AviaShapePath%20%22%40ex%3AProjectShape~ex%3AhasMilestone%22%0A%20%20%5D%20.%0A%0A%3C%23VirtualMilestoneTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AResource%20%3B%0A%20%20st%3Ashape%20ex%3AMilestoneShape%20%3B%0A%20%20st%3Areferences%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20%3C%23VirtualTaskTree%3E%20%3B%0A%20%20%20%20st%3AviaPredicate%20ex%3AhasTask%0A%20%20%5D%20%2C%0A%20%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20%3C%23VirtualIssueTree%3E%20%3B%0A%20%20%20%20st%3AviaPredicate%20ex%3AhasIssue%0A%20%20%5D%20.%0A%0A%3C%23VirtualTaskTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AResource%20%3B%0A%20%20st%3Ashape%20ex%3ATaskShape%20%3B%0A%20%20st%3Areferences%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20%3C%23NonRDFResourceTree%3E%20%3B%0A%20%20%20%20st%3AviaShapePath%20%22%40ex%3ATaskShape~ex%3AhasAttachment%22%0A%20%20%5D%20.%0A%0A%3C%23VirtualIssueTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3AIssueShape%20%3B%0A%20%20st%3Areferences%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20%3C%23NonRDFResourceTree%3E%20%3B%0A%20%20%20%20st%3AviaShapePath%20%22%40ex%3AIssueShape~ex%3AhasAttachment%22%0A%20%20%5D%20.%0A%0A%3C%23NonRDFResourceTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3ANonRDFResource%20.%0A%0A&manifestURL=http%3A%2F%2Fshex.io%2Fwebapps%2Fshex.js%2Fexamples%2Fmanifest.json&shape-map=%3C%23ProjectTree%3E%40%3C%23ShapeTreeShape%3E&interface=human&success=proof&regexpEngine=eval-threaded-nerr).

``` turtle
<#NonRDFResourceTree>
  a st:ShapeTree ;
  st:expectsType st:NonRDFResource .
```

I suspect these should be `st:RDFResources` but you can probably record the intended structure faster than i can invent it.

Reply:
That was intentional - this is one of three reserved shape tree types that are included in the vocabulary at shapetrees.ttl:

```turtle
################################################################
# Resource Type Shape Tree Definitions
################################################################

:NonRDFResourceTree
    a :ShapeTree ;
    :expectsType :NonRDFResource .

:ResourceTree
    a :ShapeTree ;
    :expectsType :Resource .

:ContainerTree
    a :ShapeTree ;
    :expectsType :Container .
```

Reply:
Oh right. needs an '@'

Reply:
... or needs `<#ShapeTreeShape>` to be a [disjunction of a shape and a value set](http://shex.io/webapps/shex.js/doc/shex-simple.html?schema=BASE%20%3Chttp%3A%2F%2Fwww.w3.org%2Fns%2Fshapetrees%3E%0APREFIX%20st%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2Fns%2Fshapetrees%23%3E%0APREFIX%20rdfs%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0APREFIX%20xsd%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23%3E%0APREFIX%20skos%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2004%2F02%2Fskos%2Fcore%23%3E%0A%0A%3C%23ShapeTreeShape%3E%20%7B%0A%20%20a%20%5Bst%3AShapeTree%5D%20%3B%0A%20%20(%0A%20%20%20%20st%3AexpectsType%20%5Bst%3AContainer%5D%20%3B%0A%20%20%20%20st%3Acontains%20%40%3C%23ShapeTreeShape%3E%20*%20%3B%0A%20%20%20%20%7C%0A%20%20%20%20st%3AexpectsType%20%5Bst%3AResource%20st%3ANonRDFResource%5D%0A%20%20)%20%3B%0A%20%20rdfs%3Alabel%20xsd%3Astring%20%3F%20%3B%0A%20%20st%3Areferences%20%40%3C%23ReferenceShape%3E%20*%20%3B%0A%20%20st%3Ashape%20IRI%20%3F%0A%7D%20OR%20%5Bst%3ANonRDFResourceTree%5D%0A%0A%3C%23ReferenceShape%3E%20%7B%0A%20%20st%3AreferencesShapeTree%20IRI%20%3B%0A%20%20(%20st%3AviaShapePath%20xsd%3Astring%0A%20%20%20%20%7C%0A%20%20%20%20st%3AviaPredicate%20IRI%20)%0A%7D%0A%0A%3C%23ManagerShape%3E%20%7B%0A%20%20a%20%5Bst%3AManager%5D%20%3B%0A%20%20st%3AhasAssignment%20%40%3C%23AssignmentShape%3E%20%2B%0A%7D%0A%0A%3C%23AssignmentShape%3E%20%7B%0A%20%20a%20%5Bst%3AAssignment%5D%20%3B%0A%20%20st%3Aassigns%20IRI%20%3B%0A%20%20st%3Amanages%20IRI%20%3B%0A%20%20st%3AhasRootAssignment%20IRI%20%3B%0A%20%20(%20st%3AfocusNode%20IRI%20%3B%0A%20%20%20%20st%3Ashape%20IRI%20)%20%3F%0A%7D%0A%0A%3C%23DescriptionSetShape%3E%20%7B%0A%20%20a%20%5Bst%3ADescriptionSet%5D%20%3B%0A%20%20st%3AusesLanguage%20xsd%3Alanguage%0A%7D%0A%0A%3C%23DescriptionShape%3E%20%7B%0A%20%20a%20%5Bst%3ADescription%5D%20%3B%0A%20%20st%3AinDescriptionSet%20%40%3C%23DescriptionSetShape%3E%20%3B%0A%20%20st%3Adescribes%20IRI%20%3B%0A%20%20skos%3AprefLabel%20xsd%3Astring%20%3B%0A%20%20skos%3Adefinition%20xsd%3Astring%20%3F%20%3B%0A%20%20skos%3Anarrower%20IRI%20%3F%20%3B%0A%20%20skos%3Abroader%20IRI%3F%20%3B%0A%7D&data=BASE%20%3Chttp%3A%2F%2Fa.example%2Fstz%3E%0APREFIX%20st%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2Fns%2Fshapetrees%23%3E%0APREFIX%20rdfs%3A%20%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0APREFIX%20ex%3A%20%3Chttp%3A%2F%2Fwww.example.com%2Fns%2Fex%23%3E%0A%0A%3C%3E%20st%3AdescribedBy%20ex%3Aproject-en%2C%20ex%3Aproject-es%2C%20ex%3Aproject-nl%2C%20ex%3Aproject-ko%20.%0A%0A%3C%23ProjectTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3AProjectShape%20%3B%0A%20%20st%3Acontains%20%3C%23MilestoneTree%3E%20.%0A%0A%3C%23MilestoneTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3AMilestoneShape%20%3B%0A%20%20st%3Acontains%20%3C%23TaskTree%3E%2C%20%3C%23IssueTree%3E%20.%0A%0A%3C%23TaskTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3ATaskShape%20%3B%0A%20%20st%3Acontains%20st%3ANonRDFResourceTree%20.%0A%0A%3C%23IssueTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3AIssueShape%20%3B%0A%20%20st%3Acontains%20st%3ANonRDFResourceTree%20.%0A%0A%3C%23VirtualProjectTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AResource%20%3B%0A%20%20st%3Ashape%20ex%3AProjectShape%20%3B%0A%20%20st%3Areferences%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20%3C%23VirtualMilestoneTree%3E%20%3B%0A%20%20%20%20st%3AviaShapePath%20%22%40ex%3AProjectShape~ex%3AhasMilestone%22%0A%20%20%5D%20.%0A%0A%3C%23VirtualMilestoneTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AResource%20%3B%0A%20%20st%3Ashape%20ex%3AMilestoneShape%20%3B%0A%20%20st%3Areferences%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20%3C%23VirtualTaskTree%3E%20%3B%0A%20%20%20%20st%3AviaPredicate%20ex%3AhasTask%0A%20%20%5D%20%2C%0A%20%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20%3C%23VirtualIssueTree%3E%20%3B%0A%20%20%20%20st%3AviaPredicate%20ex%3AhasIssue%0A%20%20%5D%20.%0A%0A%3C%23VirtualTaskTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AResource%20%3B%0A%20%20st%3Ashape%20ex%3ATaskShape%20%3B%0A%20%20st%3Areferences%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20st%3ANonRDFResourceTree%20%3B%0A%20%20%20%20st%3AviaShapePath%20%22%40ex%3ATaskShape~ex%3AhasAttachment%22%0A%20%20%5D%20.%0A%0A%3C%23VirtualIssueTree%3E%0A%20%20a%20st%3AShapeTree%20%3B%0A%20%20st%3AexpectsType%20st%3AContainer%20%3B%0A%20%20st%3Ashape%20ex%3AIssueShape%20%3B%0A%20%20st%3Areferences%20%5B%0A%20%20%20%20st%3AreferencesShapeTree%20st%3ANonRDFResourceTree%20%3B%0A%20%20%20%20st%3AviaShapePath%20%22%40ex%3AIssueShape~ex%3AhasAttachment%22%0A%20%20%5D%20.%0A%0A%0A&manifestURL=http%3A%2F%2Fshex.io%2Fwebapps%2Fshex.js%2Fexamples%2Fmanifest.json&shape-map=%3C%23ProjectTree%3E%40%3C%23ShapeTreeShape%3E&interface=human&success=proof&regexpEngine=eval-threaded-nerr). I think that captures the intention.

Reply:
Resolved in https://github.com/shapetrees/specification/pull/71/commits/40e7685b4a7bebfa7c5b72585d1d769e8748664d